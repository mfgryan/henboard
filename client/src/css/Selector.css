#Selector{
    margin: 20px 5px 0 5px;
}
