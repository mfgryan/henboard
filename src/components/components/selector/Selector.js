// react dep
import React from "react";

// bootstrap dep
import { Col, Row } from "react-bootstrap";

const Selector = ( { sprints, setSprint } ) => {
    return (
        <select onChange={setSprint} value={sprints.find(sprint => sprint.current).week} >        
            {sprints.map((sprint) => 
                <option value={sprint.week} key={sprint.week} >{sprint.week}</option>
            )}
        </select>
    )
}

export default Selector; 
