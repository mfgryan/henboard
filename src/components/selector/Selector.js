// react dep
import React from "react";

// bootstrap dep
import { Col, Row, DropdownButton, MenuItem } from "react-bootstrap";

const Selector = ( { project, sprints, currentSprint, setSprint, addSprint } ) => {
    const title = <span>Sprint Week {currentSprint.week}</span>
    return (
        <Row>
            <Col md={12}>
                <DropdownButton noCaret onSelect={(event) => setSprint(event,project)} id={project} title={title}>
                    {sprints.map((sprint) =>
                        <MenuItem key={sprint.week} eventKey={sprint.week}>{sprint.week}</MenuItem>
                    )}
                    <MenuItem divider />
                    <MenuItem eventKey={"add"}>New Sprint + </MenuItem>
                </DropdownButton>
            </Col>
        </Row>
    )
}

export default Selector; 
