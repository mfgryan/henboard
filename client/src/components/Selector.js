// react dep
import React, { Component } from "react";
import { Col, Row, DropdownButton, MenuItem } from "react-bootstrap";

// style dep
import "../css/Selector.css";

class Selector extends Component { 
    constructor(props){
        super(props);
        this.state = {showAll: false};
    }
    render() {
        let range = this.state.showAll ? this.props.items.length : 3 ;
        return (
            <Row className="Selector">
                <Col md={12}>
                    <DropdownButton noCaret onSelect={(event) => this.props.itemClicked(event)} id="Selector" title={this.props.title}>
                        {this.props.items.reverse().slice(0,range).map((item, index) =>
                            <MenuItem key={item[this.props.itemKey]} eventKey={item[this.props.itemKey]}>
                                {this.props.items.length-index}&nbsp;&nbsp;<small>{item[this.props.itemKey]}</small>
                            </MenuItem>
                        )}
                        {!this.state.showAll && 
                            <li>
                                <a href="#" onClick={() => this.setState({showAll: true})}><b>load all</b></a>
                            </li>}
                        <MenuItem divider />
                        <MenuItem eventKey={"add"}>{this.props.addItemTitle}</MenuItem>
                    </DropdownButton>
                </Col>
            </Row>
            )
    }
}

export default Selector; 
