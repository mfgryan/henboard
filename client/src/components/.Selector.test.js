import React from "react";
import Selector from "./Selector";
import renderer from "react-test-renderer";

it("Selector renders correctly", () => {
    let items = [
        {
            week: "01/01/01"
        }
    ];
    const tree = renderer
        .create(
            <Selector
                title="foo"
                items={items}
                addItemTitle="bar"
                itemKey="week"
                itemClicked={() => {}}>
                test
            </Selector>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
