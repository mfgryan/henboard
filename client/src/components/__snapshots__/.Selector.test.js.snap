exports[`test Selector renders correctly 1`] = `
<div
  className="Selector row">
  <div
    className="col-md-12">
    <div
      className="dropdown btn-group">
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-default"
        disabled={false}
        id="Selector"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button">
        foo
      </button>
      <ul
        aria-labelledby="Selector"
        className="dropdown-menu"
        role="menu">
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1">
            1
              
            <small>
              01/01/01
            </small>
          </a>
        </li>
        <li
          onKeyDown={[Function]}
          onSelect={[Function]}>
          <a
            href="#"
            onClick={[Function]}>
            <b>
              load all
            </b>
          </a>
        </li>
        <li
          className="divider"
          onKeyDown={[Function]}
          role="separator"
          style={undefined} />
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1">
            bar
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
